bpf/process/bpf_execve_bprm_commit_creds.c

```diff
@@ -102,8 +102,16 @@ BPF_KPROBE(tg_kp_bprm_committing_creds, struct linux_binprm *bprm)
 			heap->info.secureexec |= EXEC_FS_SETUID;
 	}
 	/* Is setgid? */
-	if (egid != gid)
+	if (egid != gid) {
 		heap->info.secureexec |= EXEC_SETGID;
+		/* Is egid is being changed to real root? */
+		gid = BPF_CORE_READ(bprm, cred, gid.val);
```

Comment:
Same question as for uid, what is `bprm->cred.gid`? Can it be different than task->cred.gid?

Reply:
Yes it can be different at the time of our hook, see this comment https://github.com/cilium/tetragon/pull/1786#discussion_r1447233686 same thing point 3